pallets/loans/src/entities/pricing/external.rs

```diff
@@ -126,7 +130,12 @@ impl<T: Config> ExternalActivePricing<T> {
 	}
 
 	pub fn outstanding_principal(&self, pool_id: T::PoolId) -> Result<T::Balance, DispatchError> {
-		let price = T::PriceRegistry::get(&self.info.price_id, &pool_id)?.0;
+		let price = match T::PriceRegistry::get(&self.info.price_id, &pool_id) {
+			Ok(data) => Ok(data.0),
+			Err(_) => self
+				.latest_settlement_price
+				.ok_or(DispatchError::Other("No price found")),
```

Comment:
Theoretically, this can never happen, isn't it? That's the reason not to use a defined error, right? If that's the case, I would append to the message that "this can not happen" or something like that.

Reply:
Actually, I'm pretty sure we can not have an active loan without previously having set a `latest_settlement_price` so I think this variable is not needed to be an `Option` if we initialize the `ExternalActivePricing` struct with the `settlement_price`

Reply:
Done here: [ae20e0c](https://github.com/centrifuge/centrifuge-chain/pull/1512/commits/ae20e0cbd83437e140b49181cb9af5642d2541cc)